src/System.Linq.Expressions/tests/IndexExpression/IndexExpressionVisitorTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections.Generic;
+using System.Reflection;
+using Xunit;
+
+namespace System.Linq.Expressions.Tests.IndexExpression
+{
+    public class IndexExpressionVisitorTests
+    {
+        private class IndexVisitor : ExpressionVisitor
+        {
+            public IndexVisitor(Expressions.IndexExpression expr, Expression newObject, Expression[] newArguments)
+            {
+                _dict = new Dictionary<Expression, Expression>
+                {
+                    {expr.Object, newObject}
+                };
+
+                for (int i = 0; i < expr.Arguments.Count; i++)
+                {
+                    _dict.Add(expr.Arguments[i], newArguments?[i]);
+                }
+            }
+
+            private readonly Dictionary<Expression, Expression> _dict;
+
+            public override Expression Visit(Expression node)
+            {
+                return _dict.ContainsKey(node)
+                    ? _dict[node]
+                    : base.Visit(node);
```

Comment:
Nit: perf doesn't matter so much in these tests, but this would be better as:
```C#
Expression result;
return _dict.TryGetValue(node, out result) ? result : base.Visit(node);
```
so as to avoid the second lookup.